lms/djangoapps/commerce/management/commands/create_orders_for_old_enterprise_course_enrollment.py

```diff
@@ -0,0 +1,261 @@
+"""
+Management command to
+./manage.py lms create_orders_for_old_enterprise_course_enrollment
+./manage.py lms create_orders_for_old_enterprise_course_enrollment --start-index=0 --end-index=100
+./manage.py lms create_orders_for_old_enterprise_course_enrollment --start-index=0 --end-index=100 --batch-size=20
+"""
+
+import traceback
+from textwrap import dedent
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+
+from django.core.management.base import BaseCommand, CommandError
+from opaque_keys.edx.keys import CourseKey
+from requests import Timeout
+from slumber.exceptions import SlumberBaseException
+
+from student.models import CourseEnrollment
+from openedx.core.djangoapps.commerce.utils import ecommerce_api_client
+from enterprise.models import EnterpriseCourseEnrollment
+
+from util.query import use_read_replica_if_available
+
+User = get_user_model()
+
+
+class Command(BaseCommand):
+    """
+    Command to back-populate orders(in e-commerce) for the enterprise_course_enrollments.
+    """
+    help = dedent(__doc__).strip()
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        service_user = User.objects.get(username=settings.ECOMMERCE_SERVICE_WORKER_USERNAME)
+        self.client = ecommerce_api_client(service_user)
+
+    def _get_enrollments_queryset(self, start_index, end_index):
+        """
+
+        Args:
+            start_index: start index or None
+            end_index:  end index or None
+
+        Returns:
+            EnterpriseCourseEnrollments Queryset
+
+        """
+        self.stdout.write(u'Getting enrollments from {start} to {end} index (as per command params)'
+                          .format(start=start_index or 'start', end=end_index or 'end'))
+        enrollments_qs = EnterpriseCourseEnrollment.objects.filter(
```

Comment:
we should modify the filter to exclude audit enrollments here instead of filter them later.

Reply:
`mode` of the enrollment is actually in another table `CourseEnrollment` which does not have any relation to `EnterpriseCourseEnrollment` so we had to filter it in separate query as I did.